docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+    * [Repeat Event](#repeat-event)
+    * [Changing status of events](#changing-status-of-events)
+    * [Help Command](#help-command)
+    * [Printing calendar](#printing-calendar)
+    * [Checking schedule availability](#checking-schedule-availability)
+    * [Adding deadline to event](#adding-deadline-to-event)
+    * [Setting reminder](#setting-reminder)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Setting up, getting started
+1. Fork the Scheduler--; repo from this [link](https://github.com/AY2021S1-CS2113T-T12-4/tp).
+2. Clone the fork on to your computer.
+3. Open Intellij. If you are not in the welcome screen, click `File` -> 'Close project' 
+to close the existing project dialog. 
+4. Set up the correct JDK version for Gradle.
+    1. Click `Configure` -> `Project Defaults` -> `Project Structure`
+    2. Click `New...` and set it to the directory of the JDK.
+5. Click `Open or Import` in Intellij.
+6. Locate the `build.gradle` file, select it and click `OK`.
+7. If asked, choose to `Open as Project`.
+8. Click `OK` to accept the default settings.
+9. Wait for the importing process to finish.
+10. Locate run the 'Duke.java' file, right click and select `Run Duke.main()`.
+11. Navigate to the `test\java` folder and right click `seedu.duke` and select `Run Tests in 'seedu.duke'`.
+If the setup is successful, you should see this after step 10. All tests should also pass.
+````
+_________________________________
+Welcome to scheduler--;!
+What can I do for you?
+_________________________________
+File Created: Personal
+File Created: Zoom
+File Created: Timetable
+File Created: Goal
+The file has successfully been loaded!
+_________________________________
+You have no events today!
+_________________________________
+````
+## Design
+### Architecture
 
+![Architecture Diagram](./diagrams/Architecture.jpg)
 
-## Product scope
-### Target user profile
+The ***Architecture Diagram*** given above explains the high-level design of App. Given below is a quick overview of
+each component.
 
-{Describe the target user profile}
+`Duke` class contains the main method. It is responsible for,
 
-### Value proposition
+- At app launch: Initializes the components in the correct sequence, and connects them up with each other.  
+- At shut down: Stores the user data from memory into the hard disk.
 
-{Describe the value proposition: what problem does it solve?}
+The rest of the App consists of five components.
 
-## User Stories
+- `Ui`: The Ui of the App.
+- `Parser`: Parses the input to determine which command to execute.
+- `Command`: The command executer.
+- `UserData`: Holds the data of the App in the memory.
+- `Storage`: Reads data from, and writes data to, the hard disk.
+
+#### Ui Component
+The 'Ui' component is in charge of handling input from users and system output.
+
+#### Command Component
+
+![Diagram for commmand](./diagrams/Command.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/39303087/97659754-bfd37180-1aaa-11eb-9521-717b5e4b41fc.png)

You may want to consider using XYZCommand to represent all of these classes, it is a little lengthy.